### PR TITLE
cockroach gen docs

### LINKED ISSUE #98: gen command
`./cockroach help gen`

### LINKED ISSUE #648: Docs for generating example database
On `develop` branch as of Sept. 13.

Command is:
`cockroach gen example-data | cockroach sql`

According to @knz: "gives you 2 tables, a foreign key, some random data with nice opportunities for grouping and aggregation."

PR: https://github.com/cockroachdb/cockroach/pull/9231

Think this has some potential for improving the [Learn CockroachDB SQL](https://www.cockroachlabs.com/docs/learn-cockroachdb-sql.html) docs with a second set of more advanced instructions.

### PR DESCRIPTION
**Pull request by sploiselle**
Documents `cockroach gen`.

Closes #98
Closes #648

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/755)

### CODE REVIEW

**Comment by sploiselle** [comment]
@knz If you wouldn't mind taking a look at this, I'd really appreciate it.

File: generate-cockroach-command-resources.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- `bash` autocompletion script
+- Example data
+
+<div id="toc"></div>
+
+## Subcommands
+
+| Subcommand | Usage |
+| -----------|------ |
+| `man` | Generate man pages for CockroachDB |
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing period at end of description. Same for other subcommands.

File: generate-cockroach-command-resources.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+| `autocomplete` | Generate bash autocompletion script for CockroachDB |
+| `example-data` | Generate example SQL code |
+
+## Synopsis
+
+~~~ shell
+# Generate man pages
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+$ cockroach gen man 
+
+# Generate bash autocompletion script
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+## Flags
+
+The `gen` subcommands only support [logging flags](cockroach-commands.html#logging-flags).
+
+## Examples
+
+### Generate man Pages
+
+~~~ shell
+# Generate man pages
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+$ cockroach gen man
+
+# Move the man pages to the man directory
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+$ sudo mv man/man1/* /usr/share/man/man1
+
+# Access man pages
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+$ man cockroach
+~~~
+
+### Generate bash Autocompletion Script
+
+~~~ shell
+# Generate bash autocompletion script
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

File: generate-cockroach-command-resources.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+$ cockroach gen autocomplete
+
+# Add the script to your .bashrc and .bash_profle
```
Comment (jseldess) [inline generate-cockroach-command-resources.md]: Missing colon.

**Comment by knz** [comment]
Reviewed 1 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, 13 unresolved discussions.

---

*[generate-cockroach-command-resources.md, line 2 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTtnXFAB_Ni1VqjkNZM:-KTtnXFBMJ_aIZ09osUq:bnr2k11) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L2)):*
> ```Markdown
> ---
> title: Generate Cockroach Command Resources
> ```

I'm not fond of this title. The problem is that I can't make sense of either ((Cockroach command) resources) nor (cockroach (command resources)).
May I suggest: `Automatic generator for configuration files and examples.`

---

*[generate-cockroach-command-resources.md, line 3 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTtnvVPe-b4gew_J9Wo:-KTtnvVQcPIL47y0h0Co:b-se7mac) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L3)):*
> ```Markdown
> ---
> title: Generate Cockroach Command Resources
> summary: To generate additional resources for the cockroach command, such as man pages, use the cockroach gen command.
> ```

Summary: To generate ancillary data usable in combination with the cockroach command, such as ...

---

*[generate-cockroach-command-resources.md, line 9 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto4amq7P1ZN0uXtKY:-KTto4annnww2L_jPpCZ:bhhis79) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L9)):*
> ```Markdown
> You can generate additional resources for the `cockroach` command using `cockroach gen`, including:
> 
> - `man` pages
> ```

perhaps link to wikipedia here: https://en.wikipedia.org/wiki/Man_page

---

*[generate-cockroach-command-resources.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto0aTjpa0Fp0su8Lb:-KTto0aUu0sOOExZcsER:b-2lspdl) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L11)):*
> ```Markdown
> - `man` pages
> - `bash` autocompletion script
> - Example data
> ```

Example SQL suitable to populate test databases

---

*[generate-cockroach-command-resources.md, line 70 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto_c7pbBFcNZLlqNI:-KTto_c9zizpm_iyozuc:b18g982) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L70)):*
> ```Markdown
> ### Generate Example Data
> 
> To test out CockroachDB, you can generate an example database, `startrek`, that contains 2 tables, `episodes` and `quotes`.
> ```

We support multiple example databases. However, please just mention that we support multiple databases and that the user can retrieve a list with `help gen example-data`, but do not give any details about the `intro` database. It is meant as an easter egg.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/755)*

**Comment by sploiselle** [comment]
Review status: 0 of 3 files reviewed at latest revision, 13 unresolved discussions.

---

*[generate-cli-utilities-and-example-data.md, line 70 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto_c7pbBFcNZLlqNI:-KTu5137zUYXRvxaGu0v:b1s99i) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cli-utilities-and-example-data.md#L70)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> We support multiple example databases. However, please just mention that we support multiple databases and that the user can retrieve a list with `help gen example-data`, but do not give any details about the `intro` database. It is meant as an easter egg.
</details>
Added note. Nice easter egg :)

---

*[generate-cockroach-command-resources.md, line 2 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTtnXFAB_Ni1VqjkNZM:-KTtyM-HcOb8ORpWpwZo:b9se2cu) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L2)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> I'm not fond of this title. The problem is that I can't make sense of either ((Cockroach command) resources) nor (cockroach (command resources)).
> May I suggest: `Automatic generator for configuration files and examples.`
</details>
Changing to `Generate CLI Utilities & Example Data`

---

*[generate-cockroach-command-resources.md, line 3 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTtnvVPe-b4gew_J9Wo:-KTu-A2bD-4mMfJUX5qo:bxihxdt) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L3)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Summary: To generate ancillary data usable in combination with the cockroach command, such as ...
</details>
Reworded

---

*[generate-cockroach-command-resources.md, line 9 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto4amq7P1ZN0uXtKY:-KTu-CJ_KxTaE90TOQnq:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L9)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> perhaps link to wikipedia here: https://en.wikipedia.org/wiki/Man_page
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTto0aTjpa0Fp0su8Lb:-KTu-D50T_nttnkFVRA_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L11)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Example SQL suitable to populate test databases
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 19 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-19:-KTu-NATIbT9gV2EozT0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L19)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing period at end of description. Same for other subcommands.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-26:-KTu-Oc9zBeG-NYYVbJN:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L26)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 29 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-29:-KTu-Ox1POgoGUcsWGHz:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L29)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 45 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-45:-KTu-PRV8jR2L9hDx_Nm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L45)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 48 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-48:-KTu-PlByY1RFlgCopdY:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L48)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 51 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-51:-KTu-Q2__nCbccZfaSA6:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L51)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-58:-KTu-QQXYS5EXgPdRRRx:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L58)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---

*[generate-cockroach-command-resources.md, line 61 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/755#-KTt_Iq4XyBMljbRENtu-r1-61:-KTu-QsS8JKVylhNG6FF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f7bc4ab2c001feaa003c0a7cfff463ec21ceadde/generate-cockroach-command-resources.md#L61)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Missing colon.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/755)*

**Pull request merged**